cmd/libsnap-confine-private/tool.c

```diff
@@ -80,11 +89,18 @@ void sc_call_snap_update_ns(int snap_update_ns_fd, const char *snap_name, struct
                     "--from-snap-confine", snap_name_copy, NULL};
     char *envp[] = {"SNAPD_DEBUG=x", NULL};
 
-    /* invoke snap-update-ns having switched to root group, so that created
-     * directories, files and locks are owned by the root group. */
-    sc_identity root_identity = sc_root_group_identity();
-    sc_call_snapd_tool_with_apparmor(snap_update_ns_fd, "snap-update-ns", apparmor, aa_profile, root_identity, argv,
-                                     envp);
+    /* keep the current identity, privileges are carried over through capabilities */
+    const cap_value_t caps[] = {
```

Comment:
right now all of these are the same listed in snap-confine.c ? should we share the definition? move it inside the helper?

Reply:
I think I'd keep them here for easier auditing rather than outsourcing them to external helpers. FWIW, there is a piece of code in tool.c which lists capabilities for invocation of a specific helper, but those always need to be a subset of what's defined here.